### PR TITLE
Update Agenda Tests + Misc Fixes

### PR DESCRIPTION
**Pull request by marcbonnici**

### CODE REVIEW

File: wa/framework/execution.py [hunk 1]
```diff
@@ -227,7 +227,8 @@ def add_event(self, message):
     def take_screenshot(self, filename):
         filepath = self._get_unique_filepath(filename)
         self.tm.target.capture_screen(filepath)
-        self.add_artifact('screenshot', filepath, kind='log')
```
Comment (setrofim) [inline wa/framework/execution.py]: If `capture_screen()` failed, wouldn't that result in a `TargetError` and therefore never get to `add_artifiact` anyway?

File: wa/framework/execution.py [hunk 2]
```diff
@@ -227,7 +227,8 @@ def add_event(self, message):
     def take_screenshot(self, filename):
```
Comment (setrofim) [inline wa/framework/execution.py]: In the commit message:

"as metric" --> "as an artifact"

Comment (marcbonnici) [reply wa/framework/execution.py]: Fixed

File: wa/framework/execution.py (hunk 1 above)
Comment (marcbonnici) [reply wa/framework/execution.py]: In the case of trying to take a screenshot from a `LinuxTarget` if scot is not installed then `capture_screen()` simply returns with a debug message printed.

**Pull request merged**